File: sdk/java/recipes/ProfileLifecycle.java
// Recipe — Saved profile lifecycle (start → use → stop)
// Compile with OkHttp on classpath; set MLX_FOLDER_ID / MLX_PROFILE_ID env vars.
import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class ProfileLifecycle {
    public static void main(String[] args) throws Exception {
        String host = env("MLX_LAUNCHER_HOST", "launcher.mlx.yt");
        String port = env("MLX_LAUNCHER_PORT", "45001");
        String folderId = require("MLX_FOLDER_ID");
        String profileId = require("MLX_PROFILE_ID");
        String automation = env("MLX_AUTOMATION_TYPE", "puppeteer");
        String headless = env("MLX_HEADLESS", "false");

        String startUrl = String.format(
            "https://%s:%s/api/v2/profile/f/%s/p/%s/start?automation_type=%s&headless_mode=%s",
            host, port, folderId, profileId, automation, headless
        );
        String stopUrl = String.format("https://%s:%s/api/v1/profile/stop/p/%s", host, port, profileId);

        OkHttpClient client = new OkHttpClient();
        try {
            Request start = new Request.Builder().url(startUrl).get()
                .addHeader("Accept", "application/json").build();
            try (Response resp = client.newCall(start).execute()) {
                if (!resp.isSuccessful()) throw new RuntimeException("Start HTTP " + resp.code());
                System.out.println("Started: " + resp.body().string());
            }
        } finally {
            Request stop = new Request.Builder().url(stopUrl).get()
                .addHeader("Accept", "application/json").build();
            try (Response resp = client.newCall(stop).execute()) {
                if (!resp.isSuccessful()) throw new RuntimeException("Stop HTTP " + resp.code());
                System.out.println("Stopped profile " + profileId);
            }
        }
    }

    private static String env(String key, String fallback) {
        String v = System.getenv(key);
        return v != null && !v.isBlank() ? v : fallback;
    }

    private static String require(String key) {
        String v = System.getenv(key);
        if (v == null || v.isBlank()) throw new IllegalStateException("Set " + key);
        return v;
    }
}

File: sdk/csharp/recipes/ProfileLifecycle.cs
// Recipe — Saved profile lifecycle (start → use → stop)
// dotnet script sdk/csharp/recipes/ProfileLifecycle.cs
using System;
using System.Net.Http;
using System.Threading.Tasks;

var host = Environment.GetEnvironmentVariable("MLX_LAUNCHER_HOST") ?? "launcher.mlx.yt";
var port = Environment.GetEnvironmentVariable("MLX_LAUNCHER_PORT") ?? "45001";
var folderId = Environment.GetEnvironmentVariable("MLX_FOLDER_ID")
    ?? throw new InvalidOperationException("Set MLX_FOLDER_ID");
var profileId = Environment.GetEnvironmentVariable("MLX_PROFILE_ID")
    ?? throw new InvalidOperationException("Set MLX_PROFILE_ID");
var automation = Environment.GetEnvironmentVariable("MLX_AUTOMATION_TYPE") ?? "puppeteer";
var headless = Environment.GetEnvironmentVariable("MLX_HEADLESS") ?? "false";

var startUrl =
    $"https://{host}:{port}/api/v2/profile/f/{folderId}/p/{profileId}/start" +
    $"?automation_type={automation}&headless_mode={headless}";
var stopUrl = $"https://{host}:{port}/api/v1/profile/stop/p/{profileId}";

using var client = new HttpClient();
try
{
    var start = new HttpRequestMessage(HttpMethod.Get, startUrl);
    start.Headers.Add("Accept", "application/json");
    var startResp = await client.SendAsync(start);
    startResp.EnsureSuccessStatusCode();
    var body = await startResp.Content.ReadAsStringAsync();
    Console.WriteLine($"Started profile {profileId}");
    Console.WriteLine(body);
    Console.WriteLine("→ attach Playwright/Selenium to CDP port from JSON, then stop");
}
finally
{
    var stop = new HttpRequestMessage(HttpMethod.Get, stopUrl);
    stop.Headers.Add("Accept", "application/json");
    var stopResp = await client.SendAsync(stop);
    stopResp.EnsureSuccessStatusCode();
    Console.WriteLine($"Stopped profile {profileId}");
}
